Pcap++/header/PcapFilter.h

```
@@ -286,19 +294,46 @@ namespace pcpp
 		 * @param[in] ipAddress The IPv4 address to build the filter with. If this address is not a valid IPv4 address an error will be
 		 * written to log and parsing this filter will fail
 		 */
-		void setAddr(const std::string& ipAddress) { m_Address = ipAddress; }
+		void setAddr(const std::string& ipAddress) { this->setAddr(IPAddress(ipAddress)); }
+
+		void setAddr(const IPAddress& ipAddress)
+		{
+			if (!ipAddress.isIPv4() && hasMask())
+			{
+				// TODO: What exception type to throw here?
+				throw std::runtime_error("Attempting to set non-IPv4 address while an IPv4 mask is set. "
```

Maybe throw `std::invalid_argument`?

---

Fixed bb69dcb